From 79629de65b6b1265d2254dbebbb8825aec21b58b Mon Sep 17 00:00:00 2001
From: Willow Barraco <contact@willowbarraco.fr>
Date: Wed, 17 Sep 2025 11:34:37 +0200
Subject: [PATCH] Fix crash with glib 2.86.0 and pygobject 3.54.2

With those recents releases, the prompt crash on:

AttributeError: 'gi.repository.Gio' object has no attribute 'UnixInputStream'. Did you mean: 'DataInputStream'?

The reason seems to be that glib and pygobject changed how they generate
platform specific APIs. Using GioUnix explicitely solve the problem.

Signed-off-by: Willow Barraco <contact@willowbarraco.fr>
---
 hiprompt_gtk/proto.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hiprompt_gtk/proto.py b/hiprompt_gtk/proto.py
index b50d2ec54db017dd5897c653e625e8ff85d1727b..01d1c87ec0099015b29dc928638b2c2d0d3e2d83 100644
--- a/hiprompt_gtk/proto.py
+++ b/hiprompt_gtk/proto.py
@@ -1,6 +1,6 @@
 import sys
 from enum import Enum
-from gi.repository import GLib, Gio
+from gi.repository import GLib, GioUnix
 from .key import Key
 
 
@@ -61,7 +61,7 @@ class Session:
         self.unlock = False
         self.status = 127
         self.mode = None
-        self.stream = Gio.UnixInputStream.new(sys.stdin.fileno(), False)
+        self.stream = GioUnix.InputStream.new(sys.stdin.fileno(), False)
         self.parse()
 
     def parse(self):
